java/arcs/core/allocator/Allocator.kt

```diff
@@ -89,6 +89,7 @@ class Allocator(
      * Start a new Arc given a [Plan] and return an [Arc].
      */
     private suspend fun startArcForPlan(plan: Plan, nameForTesting: String): Arc = mutex.withLock {
+        plan.init()
```

Comment:
Could we test is somehow? E.g. Ensure that schemas from handles and handle connections inside the Plan are registered during startArcForPlan?

Reply:
Done! Found a few bugs, I'm glad I tested.